Adding an Example For Wildcard Endpoints (#1233)
ngrok/ngrok-docs: docs/universal-gateway/http.mdx

	</TabItem>
</Tabs>

### Use multiple endpoints with wildcard subdomains

> With the [Agent CLI](/agent/cli), you can create separate endpoints for subdomains of a wildcard domain no matter where the upstream services are running.

Reviewer: ```suggestion
With the [Agent CLI](/agent/cli), you can create separate public endpoints for subdomains of a wildcard domain no matter where the upstream services are running. See [the Domains docs](/docs/universal-gateway/domains/#wildcard-domains) for more information.
```

> For example, if you reserve `*.example.com` and want to route requests to `api1.example.com` and `api2.example.com`, you can specify subdomains via the CLI and ngrok will handle routing requests to the correct endpoint, even if they're on different ports.

Reviewer: ```suggestion
For example, if you reserve `*.example.com` and want to route requests to `api1.example.com` and `api2.example.com`, you can specify subdomains via the CLI. ngrok will handle routing requests to the correct endpoint, even if they're on different ports.
```

The first one might be at port `80`:

```bash
ngrok http 80 --url https://api1.example.com
```

While the second is at port `81`:

```bash
ngrok http 81 --url https://api2.example.com
```

> 

Reviewer: ```suggestion

:::tip
You can only create public endpoints for subdomains of wildcard domains.
:::
```

> ```bash
> ngrok http 80 --domain https://api1.example.com
> ```
> 
> While the second is at port `81`:
> 
> ```bash
> ngrok http 81 --domain https://api2.example.com
> ```

Reviewer: ````suggestion
```bash
ngrok http 80 --url https://api1.example.com
```

While the second is at port `81`:

```bash
ngrok http 81 --url https://api2.example.com
```
````

	</TabItem>
</Tabs>

### Use multiple endpoints with wildcard subdomains

With the [Agent CLI](/agent/cli), you can create separate public endpoints for subdomains of a wildcard domain no matter where the upstream services are running. See [the Domains docs](/docs/universal-gateway/domains/#wildcard-domains) for more information.

For example, if you reserve `*.example.com` and want to route requests to `api1.example.com` and `api2.example.com`, you can specify subdomains via the CLI. ngrok will handle routing requests to the correct endpoint, even if they're on different ports.

The first one might be at port `80`:

```bash
ngrok http 80 --url https://api1.example.com
```

While the second is at port `81`:

```bash
ngrok http 81 --url https://api2.example.com
```

:::tip

> You can only create public endpoints for subdomains of wildcard domains.

Reviewer: ```suggestion
This is only necessary for creating _public_ endpoints to subdomains. Internal endpoints don't require a domain.
```
